From 3cfdcabc781a6ee1b9221190ae577ae57c50b729 Mon Sep 17 00:00:00 2001
From: Jiadong Zhu <Jiadong.Zhu@amd.com>
Date: Sat, 6 May 2023 17:35:05 +0800
Subject: [PATCH] ac: enable SHADOW_GLOBAL_CONFIG for preemptible ib

SHADOW_GLOBAL_CONFIG is mandatory for mid command buffer preemmption.

Fixes: 69014d8c94f (radeonsi: implement CP register shadowing)
Reviewed-by: Pierre-Eric Pelloux-Prayer <pierre-eric.pelloux-prayer@amd.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/22916>
---
 src/amd/common/ac_shadowed_regs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/amd/common/ac_shadowed_regs.c b/src/amd/common/ac_shadowed_regs.c
index 834bda34bc519..c14541efaa81a 100644
--- a/src/amd/common/ac_shadowed_regs.c
+++ b/src/amd/common/ac_shadowed_regs.c
@@ -4286,7 +4286,8 @@ void ac_create_shadowing_ib_preamble(const struct radeon_info *info,
                CC1_SHADOW_PER_CONTEXT_STATE(1) |
                CC1_SHADOW_CS_SH_REGS(1) |
                CC1_SHADOW_GFX_SH_REGS(1) |
-               CC1_SHADOW_GLOBAL_UCONFIG(1));
+               CC1_SHADOW_GLOBAL_UCONFIG(1) |
+               CC1_SHADOW_GLOBAL_CONFIG(1));
 
    if (!info->has_fw_based_shadowing) {
       for (unsigned i = 0; i < SI_NUM_SHADOWED_REG_RANGES; i++)
-- 
GitLab

